core/src/main/java/io/questdb/cairo/AbstractRecordMetadata.java

```diff
@@ -104,6 +104,11 @@ public boolean isColumnIndexed(int columnIndex) {
         return getColumnMetadata(columnIndex).isIndexed();
     }
 
+    @Override
+    public boolean isDedupKey(int columnIndex) {
```

Comment:
this method should be part of TableWriterMetadata rather than having linguine effect on the SQL subsystem

Reply:
We already have `getIndexValueBlockCapacity` in `RecordMetadata` and here even though it is a storage-only property that is not needed in SQL. 

Entirely removing these additional column storage properties from  `RecordMetadata`  and `AbstractRecordMetadata` is an unpleasant refactoring with negligible upside.